### ISSUE TITLE
Do block not detected

### ISSUE THREAD

**Issue by jrp2014** [ndmitchell/hlint#929]
The following code
```
normaliseStatic :: I.Term -> CC I.Term
normaliseStatic v@(I.Def f es) = lift $ do
  static <- isStaticFun . theDef <$> getConstInfo f
  if static then normalise v else pure v
normaliseStatic v = pure v
```
produces
```
    parse error on input `<-'
    Perhaps this statement should be within a 'do' block?
Found:
  > 
```
GHC (8.x.y) seems to compile it OK.

**Comment by zliu41** [comment]
Fixed in #931.
